packages/common/src/Utility.ts

```diff
@@ -603,3 +603,30 @@ export function parseVersionString(versionString) {
         patch: parseInt(_sp[2].split("-")[0])
     };
 }
+
+type TextSize = { width: number; height: number; };
+let g_fontSizeContext: CanvasRenderingContext2D;
+const g_fontSizeContextCache: { [key: string]: TextSize } = {};
+
+export function textSize(_text: string | string[], fontName: string = "Verdana", fontSize: number = 12, bold: boolean = false): TextSize {
+    if (!g_fontSizeContext) {
+        let fontSizeCalc = d3Select("body > #hpcc_js_font_size");
+        if (fontSizeCalc.empty()) {
+            fontSizeCalc = d3Select("body").append("canvas")
+                .attr("id", "hpcc_js_font_size")
+                ;
+        }
+        g_fontSizeContext = (fontSizeCalc.node() as HTMLCanvasElement).getContext("2d");
+    }
```

Comment:
I know this is beyond the scope of this change, but I think this if statement would be less invasive and more performant if it was:
```javascript
    if (!g_fontSizeContext) {
        g_fontSizeContext = document.createElement("canvas").getContext("2d");
    }
```

Reply:
I agree it looks a little bit "to be extra extra sure" - but performance won't be an issue as its only ever created once...